### PR TITLE
feat: Ball lightning scatters

### PR DESCRIPTION
**Pull request by scarf005**
# Summary
ball lightnings summoned by `Conjure Ball Lightning` now moves similar to foxfire. Or in other words, they wander off randomly. It has following advantages:

- accidently casting cbl on empty field is less suicidal
- from 'ball' lightning to ball 'lightning': their movement is more chaotic and interesting

# Screenshots

## After PR
Dungeon Crawl Stone Soup 0 30-a0-344-g23b102bf_01
Dungeon Crawl Stone Soup 0 30-a0-344-g23b102bf_02

the balls fly in random way, similar to foxfires
(trail is from #2809)

## Before PR
Dungeon Crawl Stone Soup 0 30-a0-341-g68270651_02

when there is no enemy in vision, the balls home into player.

### CODE REVIEW

**Comment by rawlins** [comment]
I've clicked the "approve workflow" button, the response you got in `#crawl-dev` was from someone who I guess wasn't aware of this github feature.

**Comment by scarf005** [comment]
Thank you. Could you also check #2809?

**Comment by scarf005** [comment]
strange, all the failures are from network errors...

**Comment by rawlins** [comment]
Yes, unfortunately this happens semi-regularly with CI depending on various upstream sources. LLVM in particular (which is what has errored here) is kind of unreliable, but tends to get fixed in hours-to-days when it breaks. At some point, we can trigger a rerun, it's ignorable by you for now. If you did want to run tests locally for a specific build this is possible via something like `make debug && make test` (I think I've gotten that right).

File: crawl-ref/source/spl-summoning.cc [hunk 1]
```diff
@@ -817,13 +817,16 @@ int mons_ball_lightning_hd(int pow, bool random)
...
-spret cast_conjure_ball_lightning(int pow, god_type god, bool fail)
+spret cast_conjure_ball_lightning(const actor &agent, int pow, god_type god, bool fail)
 {
     fail_check();
     bool success = false;
 
-    mgen_data cbl =_pal_data(MONS_BALL_LIGHTNING, 0, god,
-                             SPELL_CONJURE_BALL_LIGHTNING);
+    const auto att = agent.is_player() ? BEH_FRIENDLY
+                                       : SAME_ATTITUDE(agent.as_monster());
```
Comment (PleasingFungus) [inline crawl-ref/source/spl-summoning.cc:826]: I believe you can just use `BEH_COPY` here, as `_summon_data()` does.

File: crawl-ref/source/spl-summoning.cc [hunk 2]
```diff
@@ -817,13 +817,16 @@ int mons_ball_lightning_hd(int pow, bool random)
...
+    mgen_data cbl(MONS_BALL_LIGHTNING, att,
+      agent.pos(), MHITNOT, MG_FORCE_PLACE | MG_AUTOFOE);
```
Comment (PleasingFungus) [inline crawl-ref/source/spl-summoning.cc:828]: Why add `MG_FORCE_PLACE`? That seems unrelated to this PR.

File: crawl-ref/source/spl-summoning.cc [hunk 3]
```diff
@@ -817,13 +817,16 @@ int mons_ball_lightning_hd(int pow, bool random)
...
+    cbl.set_summoned(&agent, 0, SPELL_CONJURE_BALL_LIGHTNING, god);
```
Comment (PleasingFungus) [inline crawl-ref/source/spl-summoning.cc:829]: Could you just do:
```suggestion
    mgen_data cbl =_pal_data(MONS_BALL_LIGHTNING, 0, god,
                             SPELL_CONJURE_BALL_LIGHTNING);
    cbl.foe = MHITNOT;
```

Seems a little simpler to me. WDYT?

File: crawl-ref/source/spl-summoning.cc [hunk 4]
```diff
@@ -817,13 +817,16 @@ int mons_ball_lightning_hd(int pow, bool random)
     return ball_lightning_hd(pow, random) / 2;
 }
 
-spret cast_conjure_ball_lightning(int pow, god_type god, bool fail)
+spret cast_conjure_ball_lightning(const actor &agent, int pow, god_type god, bool fail)
```
Comment (PleasingFungus) [inline crawl-ref/source/spl-summoning.cc:820]: You don't need to add an agent here. This is only ever used by the player. Monster conjure ball lightning uses a completely different codepath - see https://github.com/crawl/crawl/blob/master/crawl-ref/source/mon-cast.cc#L5959 .

**Review by PleasingFungus — CHANGES_REQUESTED** [review]
Proposed changes look good! Once you fix the code per my request, I'll be happy to merge this. Apologies again for the delay.

**Comment by scarf005** [comment]
thanks for the detailed review. I've accidently deleted the fork and waiting revival, until then I'll be applying the suggestions locally.

File: crawl-ref/source/spl-summoning.cc (hunk 3 above)
Comment (scarf005) [reply crawl-ref/source/spl-summoning.cc:829]: thanks! applied to local repo.

File: crawl-ref/source/spl-summoning.cc (hunk 2 above)
Comment (scarf005) [reply crawl-ref/source/spl-summoning.cc:828]: alright, automatically resolved by applying suggested changes below.

File: crawl-ref/source/spl-summoning.cc (hunk 4 above)
Comment (scarf005) [reply crawl-ref/source/spl-summoning.cc:820]: thanks! removed the agent from parameter.

File: crawl-ref/source/spl-summoning.cc (hunk 1 above)
Comment (scarf005) [reply crawl-ref/source/spl-summoning.cc:826]: thanks, will also mark as resolved, since [suggestion from last converstation](https://github.com/crawl/crawl/pull/2813#discussion_r1063960530) fixes it.

**Comment by scarf005** [comment]
I've fixed commits accordingly with code review. however because I've accidently deleting my fork of crawl, I am unable to update the PR. should I wait until fork revival (might take a few days), or close those PRs (#2809, #2813),  make new fork, and open another PR with identical changes? which way would you prefer?

**Comment by PleasingFungus** [comment]
Either way is fine by me. Whatever's easier for you.

**Comment by scarf005** [comment]
closing in order to reopen PR from another fork.
